metals/src/main/scala/scala/meta/metals/ScalametaServices.scala

```diff
@@ -87,13 +89,16 @@ class MetalsServices(
     new SquiggliesProvider(configurationPublisher, cwd)
   val scalacProvider = new ScalacProvider
   val interactiveSemanticdbs: Observable[semanticdb.Database] =
-    sourceChangePublisher
-      .debounce(FiniteDuration(1, "s"))
-      .flatMap { input =>
-        Observable
-          .fromIterable(Semanticdbs.toSemanticdb(input, scalacProvider))
-          .executeOn(presentationCompilerScheduler)
-      }
+    if (latestConfig().scalac.completions.enabled ||
```

Comment:
I would move this condition into the `flatMap` so that we can respond to dynamic updates on the configuration. I think otherwise `val interactiveSemanticdbs` will always be `Observable.empty`

Reply:
Done in 15398a1